bc-backend/src/main/java/org/jboss/da/bc/backend/impl/FinalizerImpl.java

```diff
@@ -92,7 +92,13 @@ public Integer createBCs(int id, ProjectHiearchy toplevelBc) throws Communicatio
             BuildConfiguration bc;
             ProjectDetail project = hiearchy.getProject();
             if (project.isUseExistingBc()) {
-                List<BuildConfiguration> existingBcs = bcFinder.lookupBcByScm(project.getScmUrl(), project.getScmRevision());
+                List<BuildConfiguration> existingBcs;
+                if (project.getInternalScmUrl() == null) {
+                    existingBcs = bcFinder.lookupBcByScm(project.getExternalScmUrl(), project.getExternalScmRevision());
+                }
+                else {
+                    existingBcs = bcFinder.lookupBcByScm(project.getInternalScmUrl(), project.getInternalScmRevision());
+                }
```

Comment:
create bcFinder.lookupBcByInternalSCM and bcFinder.lookupBcByExternalSCM and combine the lists